### PR TITLE
Updated CamemBERT model card to new standardized format

### PR DESCRIPTION
**Pull request by MShaheerMalik77**
# What does this PR do?

This PR updates the CamemBERT model card `camembert.md`, by standardizing its format according to https://github.com/huggingface/transformers/issues/36979. It includes example code for performing masked language modelling with the pipeline and AutoModel classes, as well as a Quantization example.

## Before submitting
- [x] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [x] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/main/CONTRIBUTING.md#create-a-pull-request),
      Pull Request section?
- [x] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link
      to it if that's the case.
- [x] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/transformers/tree/main/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/main/docs#writing-source-documentation).
- [ ] Did you write any new necessary tests?

## Who can review?

@stevhliu please do let me know if you'd like me to make any changes!

### CODE REVIEW

File: docs/source/en/model_doc/camembert.md [hunk 1]
```diff
@@ -1,141 +1,79 @@
...
-
--->
-
-# CamemBERT
-
-<div class="flex flex-wrap space-x-1">
-<img alt="PyTorch" src="https://img.shields.io/badge/PyTorch-DE3412?style=flat&logo=pytorch&logoColor=white">
-<img alt="TensorFlow" src="https://img.shields.io/badge/TensorFlow-FF6F00?style=flat&logo=tensorflow&logoColor=white">
-<img alt="SDPA" src="https://img.shields.io/badge/SDPA-DE3412?style=flat&logo=pytorch&logoColor=white">
+# CamemBERT Base
```
Comment (stevhliu) [inline docs/source/en/model_doc/camembert.md]:
```suggestion
# CamemBERT
```

File: docs/source/en/model_doc/camembert.md [hunk 2]
```diff
@@ -1,141 +1,79 @@
-<!--Copyright 2020 The HuggingFace Team. All rights reserved.
```
Comment (stevhliu) [inline docs/source/en/model_doc/camembert.md]: No need to remove the copyright/license info

File: docs/source/en/model_doc/camembert.md [hunk 3]
```diff
@@ -1,141 +1,79 @@
...
--->
-
-# CamemBERT
-
-<div class="flex flex-wrap space-x-1">
-<img alt="PyTorch" src="https://img.shields.io/badge/PyTorch-DE3412?style=flat&logo=pytorch&logoColor=white">
-<img alt="TensorFlow" src="https://img.shields.io/badge/TensorFlow-FF6F00?style=flat&logo=tensorflow&logoColor=white">
-<img alt="SDPA" src="https://img.shields.io/badge/SDPA-DE3412?style=flat&logo=pytorch&logoColor=white">
+# CamemBERT Base
+<div style="float: right;">
```
Comment (stevhliu) [inline docs/source/en/model_doc/camembert.md:17]: Missing the SDPA badge

File: docs/source/en/model_doc/camembert.md [hunk 4]
```diff
@@ -1,141 +1,79 @@
...
-
-## Resources
-
-- [Text classification task guide](../tasks/sequence_classification)
-- [Token classification task guide](../tasks/token_classification)
-- [Question answering task guide](../tasks/question_answering)
-- [Causal language modeling task guide](../tasks/language_modeling)
-- [Masked language modeling task guide](../tasks/masked_language_modeling)
-- [Multiple choice task guide](../tasks/multiple_choice)
+[CamemBERT](https://huggingface.co/docs/transformers/model_doc/camembert) is a language model based on RoBERTa, but it was trained specifically on French text from the OSCAR dataset.
```
Comment (stevhliu) [inline docs/source/en/model_doc/camembert.md]:
```suggestion
[CamemBERT](https://huggingface.co/papers/1911.03894) is a language model based on [RoBERTa](./roberta), but trained specifically on French text from the OSCAR dataset, making it more effective for French language tasks.
```

File: docs/source/en/model_doc/camembert.md [hunk 5]
```diff
@@ -1,141 +1,79 @@
...
 
-## CamembertConfig
+What sets CamemBERT apart is that it learned from a huge, high quality collection of French data, as opposed to mixing lots of languages. This helps it really understand French better than many multilingual models.
```
Comment (stevhliu) [inline docs/source/en/model_doc/camembert.md:29]:
```suggestion
```

File: docs/source/en/model_doc/camembert.md [hunk 6]
```diff
@@ -1,141 +1,79 @@
...
 
-[[autodoc]] CamembertConfig
+Common applications of CamemBERT include masked language modeling (Fill-mask prediction), text classification (sentiment analysis), token classification (entity recognition) and sentence pair classification (entailment tasks).
```
Comment (stevhliu) [inline docs/source/en/model_doc/camembert.md:31]:
```suggestion
```

File: docs/source/en/model_doc/camembert.md [hunk 7]
```diff
@@ -1,141 +1,79 @@
...
 
-## CamembertTokenizer
+You can find all the original CamemBERT checkpoints under the [CamemBERT](https://huggingface.co/models?search=camembert) collection.
```
Comment (stevhliu) [inline docs/source/en/model_doc/camembert.md]:
```suggestion
You can find all the original CamemBERT checkpoints under the [ALMAnaCH](https://huggingface.co/almanach/models?search=camembert) organization.
```

File: docs/source/en/model_doc/camembert.md [hunk 8]
```diff
@@ -1,141 +1,79 @@
...
 
-[[autodoc]] CamembertTokenizer
-    - build_inputs_with_special_tokens
-    - get_special_tokens_mask
-    - create_token_type_ids_from_sequences
-    - save_vocabulary
+> [!TIP]
+> This model was contributed by the [Facebook AI](https://huggingface.co/facebook) team.
```
Comment (stevhliu) [inline docs/source/en/model_doc/camembert.md]:
```suggestion
> This model was contributed by the [ALMAnaCH (Inria)](https://huggingface.co/almanach) team.
```

File: docs/source/en/model_doc/camembert.md [hunk 9]
```diff
@@ -1,141 +1,79 @@
...
+>
+> Click on the CamemBERT models in the right sidebar for more examples of how to apply CamemBERT to different NLP tasks.
 
-## CamembertTokenizerFast
+The examples below demonstrate how to perform masked language modeling with `pipeline` or the `AutoModel` class.
```
Comment (stevhliu) [inline docs/source/en/model_doc/camembert.md]:
```suggestion
The examples below demonstrate how to predict the `<mask>` token with [`Pipeline`], [`AutoModel`], and from the command line.
```

File: docs/source/en/model_doc/camembert.md [hunk 10]
```diff
@@ -1,141 +1,79 @@
...
-[[autodoc]] CamembertTokenizerFast
+<hfoptions id="usage">
 
-<frameworkcontent>
-<pt>
+<hfoption id="Pipeline">
 
-## CamembertModel
+```python
+from transformers import pipeline
```
Comment (stevhliu) [inline docs/source/en/model_doc/camembert.md:48]:
```py
import torch
from transformers import pipeline

pipeline = pipeline("fill-mask", model="camembert-base", torch_dtype=torch.float16, device=0)
pipeline("Le camembert est un délicieux fromage <mask>.")
```

File: docs/source/en/model_doc/camembert.md [hunk 11]
```diff
@@ -1,141 +1,79 @@
...
 
-## CamembertForCausalLM
+</hfoption>
 
-[[autodoc]] CamembertForCausalLM
+<hfoption id="AutoModel">
 
-## CamembertForMaskedLM
+```python
+from transformers import AutoTokenizer, AutoModelForMaskedLM
```
Comment (stevhliu) [inline docs/source/en/model_doc/camembert.md:59]:
```py
import torch
from transformers import AutoTokenizer, AutoModelForMaskedLM

tokenizer = AutoTokenizer.from_pretrained("camembert-base")
model = AutoModelForMaskedLM.from_pretrained("camembert-base", torch_dtype="auto", device_map="auto", attn_implementation="sdpa")
inputs = tokenizer("Le camembert est un délicieux fromage <mask>.", return_tensors="pt").to("cuda")

with torch.no_grad():
    outputs = model(**inputs)
    predictions = outputs.logits

masked_index = torch.where(inputs['input_ids'] == tokenizer.mask_token_id)[1]
predicted_token_id = predictions[0, masked_index].argmax(dim=-1)
predicted_token = tokenizer.decode(predicted_token_id)

print(f"The predicted token is: {predicted_token}")
```

File: docs/source/en/model_doc/camembert.md [hunk 12]
```diff
@@ -1,141 +1,79 @@
...
-## CamembertForSequenceClassification
+inputs = tokenizer("Le camembert est un délicieux fromage <mask>.", return_tensors="pt")
+outputs = model(**inputs)
+```
 
-[[autodoc]] CamembertForSequenceClassification
+</hfoption>
 
-## CamembertForMultipleChoice
+</hfoptions>
```
Comment (stevhliu) [inline docs/source/en/model_doc/camembert.md]: Add an example of using the model from the command line

File: docs/source/en/model_doc/camembert.md [hunk 13]
```diff
@@ -1,141 +1,79 @@
...
 
-[[autodoc]] CamembertForMultipleChoice
+Quantization reduces the memory burden of large models by representing weights in lower precision. Refer to the [Quantization](https://huggingface.co/docs/transformers/main/en/quantization) overview for available options.
```
Comment (stevhliu) [inline docs/source/en/model_doc/camembert.md]:
```suggestion
Quantization reduces the memory burden of large models by representing weights in lower precision. Refer to the [Quantization](../quantization/overview) overview for available options.
```

File: docs/source/en/model_doc/camembert.md [hunk 14]
```diff
@@ -1,141 +1,79 @@
...
+The example below uses [BitsAndBytes](https://huggingface.co/docs/transformers/main/en/perf_infer_gpu_one#load-in-8bit-or-4bit-using-bitsandbytes) quantization to load the model in 8-bit precision.
```
Comment (stevhliu) [inline docs/source/en/model_doc/camembert.md]:
```suggestion
The example below uses [bitsandbytes](../quantization/bitsandbytes) quantization to quantize the weights to 8-bits.
```

File: docs/source/en/model_doc/camembert.md [hunk 15]
```diff
@@ -1,141 +1,79 @@
...
 
-## CamembertForTokenClassification
+```python
+from transformers import AutoTokenizer, AutoModelForMaskedLM, BitsAndBytesConfig
```
Comment (stevhliu) [inline docs/source/en/model_doc/camembert.md:93]: Show an end-to-end example here with the large version of the model https://huggingface.co/almanach/camembert-large

File: docs/source/en/model_doc/camembert.md [hunk 16]
```diff
@@ -1,141 +1,79 @@
...
+quant_config = BitsAndBytesConfig(load_in_8bit=True)
+model = AutoModelForMaskedLM.from_pretrained(
+    "camembert-base",
+    quantization_config=quant_config,
+    device_map="auto"
+)
+tokenizer = AutoTokenizer.from_pretrained("camembert-base")
+```
+	
+## Notes
```
Comment (stevhliu) [inline docs/source/en/model_doc/camembert.md]: You can remove these notes

File: docs/source/en/model_doc/camembert.md [hunk 17]
```diff
@@ -1,141 +1,79 @@
...
-
-- [Text classification task guide](../tasks/sequence_classification)
-- [Token classification task guide](../tasks/token_classification)
-- [Question answering task guide](../tasks/question_answering)
-- [Causal language modeling task guide](../tasks/language_modeling)
-- [Masked language modeling task guide](../tasks/masked_language_modeling)
-- [Multiple choice task guide](../tasks/multiple_choice)
+[CamemBERT](https://huggingface.co/docs/transformers/model_doc/camembert) is a language model based on RoBERTa, but it was trained specifically on French text from the OSCAR dataset.
 
-## CamembertConfig
```
Comment (stevhliu) [inline docs/source/en/model_doc/camembert.md:61]: Add all the API references back, they don't need to be removed

File: docs/source/en/model_doc/camembert.md [hunk 18]
```diff
@@ -1,141 +1,79 @@
...
-## TFCamembertForQuestionAnswering
-
-[[autodoc]] TFCamembertForQuestionAnswering
-
-</tf>
-</frameworkcontent>
+## Resources
 
+- [Original Paper](https://arxiv.org/abs/1911.03894)
+- [Hugging Face Model Card](https://huggingface.co/camembert-base)
```
Comment (stevhliu) [inline docs/source/en/model_doc/camembert.md]: Remove this section

**Review by stevhliu — COMMENTED** [review]
Good start!

**Comment by MShaheerMalik77** [comment]
Hi @stevhliu! I've added all the suggested changes to the CamemBERT model card, please let me know if there's anything else you'd like me to update!

File: docs/source/en/model_doc/camembert.md [hunk 19]
```diff
@@ -13,80 +13,83 @@ specific language governing permissions and limitations under the License.
...
 
 -->
-
 # CamemBERT
-
-<div class="flex flex-wrap space-x-1">
-<img alt="PyTorch" src="https://img.shields.io/badge/PyTorch-DE3412?style=flat&logo=pytorch&logoColor=white">
-<img alt="TensorFlow" src="https://img.shields.io/badge/TensorFlow-FF6F00?style=flat&logo=tensorflow&logoColor=white">
-<img alt="SDPA" src="https://img.shields.io/badge/SDPA-DE3412?style=flat&logo=pytorch&logoColor=white">
+<div style="float: right;">
```
Comment (stevhliu) [inline docs/source/en/model_doc/camembert.md:17]: The badges should go above `# CamemBERT`

File: docs/source/en/model_doc/camembert.md [hunk 20]
```diff
@@ -95,13 +98,61 @@ as the information relative to the inputs and outputs.
 
 ## CamembertForMultipleChoice
 
-[[autodoc]] CamembertForMultipleChoice
+You can also use CamemBERT for multiple-choice tasks via the command line:
```
Comment (stevhliu) [inline docs/source/en/model_doc/camembert.md]: Add this to the <hfoption> blocks above and make sure the example matches that of the `Pipeline` and `AutoModel` for consistency. See the [BERT](https://huggingface.co/docs/transformers/model_doc/bert?usage=transformers+CLI) docs for an example

File: docs/source/en/model_doc/camembert.md [hunk 21]
```diff
@@ -95,13 +98,61 @@ as the information relative to the inputs and outputs.
...
+    --context "The president of France is" \
+    --choices "Emmanuel Macron" "Napoleon Bonaparte" "Marie Curie" \
+    --framework pt
+```
+
+[[autodoc]] CamembertForMultipleChoice
+  
 ## CamembertForTokenClassification
 
+You can use the `camembert/camembert-large` model for token classification tasks like Named Entity Recognition (NER).
```
Comment (stevhliu) [inline docs/source/en/model_doc/camembert.md]: This token classification example is unnecessary and can be removed

**Review by stevhliu — COMMENTED** [review]
Thanks, there are still some unresolved comments and the API references are scattered throughout the doc rather than having their own section

**Comment by MShaheerMalik77** [comment]
Hi @stevhliu. I went through all unresolved issues, applied all the suggestions and re-added the api references in the proper format. Please do let me know if there's anything you would like me to adjust!

File: docs/source/en/model_doc/camembert.md [hunk 22]
```diff
@@ -14,49 +14,108 @@ rendered properly in your Markdown viewer.
...
+predicted_token = tokenizer.decode(predicted_token_id)
 
-This implementation is the same as RoBERTa. Refer to the [documentation of RoBERTa](roberta) for usage examples as well 
-as the information relative to the inputs and outputs.
+print(f"The predicted token is: {predicted_token}")
+```
+</hfoption> 
 
-</Tip>
+<hfoption id="Command line">
```
Comment (stevhliu) [inline docs/source/en/model_doc/camembert.md]:
```suggestion
<hfoption id="transformers CLI">
```

File: docs/source/en/model_doc/camembert.md [hunk 23]
```diff
@@ -14,49 +14,108 @@ rendered properly in your Markdown viewer.
...
 
-## Resources
+```bash
```
Comment (stevhliu) [inline docs/source/en/model_doc/camembert.md:79]:
```bash
echo -e "Le camembert est un délicieux fromage <mask>." | transformers run --task fill-mask --model camembert-base --device 0
```

**Review by stevhliu — APPROVED** [review]
Thanks, just a few more things and then we can merge!

**Comment by MShaheerMalik77** [comment]
Hi @stevhliu , I have incorporated remaining suggestions to the model document, let me know if there are any further changes you'd like me to make!
